src/main/kotlin/org/koitharu/kotatsu/parsers/site/en/LineWebtoonsParser.kt

```diff
@@ -0,0 +1,426 @@
+		// this should be fine though
+		SortOrder.RATING,
+		SortOrder.UPDATED,
+	)
+	override val headers: Headers
+		get() = Headers.Builder()
+			.add("User-Agent", "nApps (Android 12;; linewebtoon; 3.1.0)")
+			.build()
+
+	override suspend fun getPageUrl(page: MangaPage): String {
+		return page.url
+	}
+
+	private suspend fun getChapters(titleNo: Long): List<MangaChapter> {
+		val firstResult = makeRequest("/lineWebtoon/webtoon/challengeEpisodeList.json?v=2&titleNo=$titleNo&startIndex=0&pageSize=30")
+
+		val totalEpisodeCount = firstResult
+			.getJSONObject("episodeList")
+			.getInt("totalServiceEpisodeCount")
+
+		val episodes = firstResult
+			.getJSONObject("episodeList")
+			.getJSONArray("episode")
+			.toJSONList()
+			.toMutableList()
+
+		while (episodes.count() < totalEpisodeCount) {
+			val page = makeRequest("/lineWebtoon/webtoon/challengeEpisodeList.json?v=2&titleNo=$titleNo&startIndex=${episodes.count()}&pageSize=30")
+				.getJSONObject("episodeList")
+				.getJSONArray("episode")
+				.toJSONList()
+
+			episodes.addAll(page)
+		}
+
+		return episodes.mapIndexed { i, jo ->
+			MangaChapter(
+				id = generateUid("$titleNo-$i"),
+				name = jo.getString("episodeTitle"),
+				number = jo.getInt("episodeSeq"),
+				url = "$titleNo-${jo.getString("episodeNo")}",
+				uploadDate = jo.getLong("modifyYmdt"),
+				// do we want to use it for anything?
+				branch = null,
+				scanlator = null,
+				source = source,
+			)
+		}.sortedBy { it.number }
+	}
+
+	override suspend fun getDetails(manga: Manga): Manga {
+		val titleNo = manga.url.toLong()
+
+		return makeRequest("/lineWebtoon/webtoon/challengeTitleInfo.json?v=2&titleNo=${titleNo}")
+			.getJSONObject("titleInfo")
+			.let { jo ->
+				Manga(
+					id = generateUid(titleNo),
+					title = jo.getString("title"),
+					altTitle = null,
+					url = "$titleNo",
+					publicUrl = "https://${domain}/en/canvas/a/list?title_no=${titleNo}",
+					rating = jo.getDouble("starScoreAverage").toFloat() / 10f,
+					isNsfw = jo.getBoolean("ageGradeNotice"),
+					coverUrl = "https://$staticDomain${jo.getString("thumbnail")}",
+					largeCoverUrl = if (jo.has("thumbnailVertical")) {
+						"https://$staticDomain${jo.getString("thumbnailVertical")}"
+					} else { null },
+					tags = setOf(parseTag(jo.getJSONObject("genreInfo"))),
+					author = jo.getString("writingAuthorName"),
+					description = jo.getString("synopsis"),
+					// I don't think the API provides this info
+					state = null,
+					chapters = getChapters(titleNo),
+					source = source,
+				)
+			}
+	}
+
+	override suspend fun getList(
+		offset: Int,
+		query: String?,
+		tags: Set<MangaTag>?,
+		sortOrder: SortOrder,
+	): List<Manga> {
+		val genre = tags.oneOrThrowIfMany()?.key ?: "ALL"
+
+		val sortOrderStr = when (sortOrder) {
+			SortOrder.UPDATED -> "UPDATE"
+			SortOrder.POPULARITY -> "READ_COUNT"
+			SortOrder.RATING -> "LIKEIT"
+			else -> {
+				throw Exception("Unreachable")
```

Comment:
Avoid throwing so generic exceptions. I suppose the `IllegalArgumentException` fits this situation  in the best way